packages/ui/src/components/pagination/pagination.tsx

```diff
@@ -1,151 +1,238 @@
+            <PaginationPrimitive.Item className="cn-pagination-item-previous">
               <PaginationPrimitive.Previous
-                size="sm"
-                onClick={handleGoToPage(currentPage > 1 ? currentPage - 1 : undefined)}
+                onClick={goToPage ? handleGoToPage(currentPage > 1 ? currentPage - 1 : undefined) : undefined}
+                href={getPageLink?.(currentPage > 1 ? currentPage - 1 : currentPage)}
                 disabled={currentPage === 1}
                 t={t}
               />
             </PaginationPrimitive.Item>
 
             {/* Pagination Items */}
-            {totalPages && (
-              <PaginationItems totalPages={totalPages} currentPage={currentPage} goToPage={handleGoToPage} />
+            {!hidePageNumbers && totalPages && (
+              <PaginationItems
+                totalPages={totalPages}
+                currentPage={currentPage}
+                getPageLink={getPageLink}
+                goToPage={goToPage ? handleGoToPage : undefined}
+                truncateLimit={truncateLimit}
+              />
             )}
 
             {/* Next Button */}
-            <PaginationPrimitive.Item>
+            <PaginationPrimitive.Item className="cn-pagination-item-next">
               <PaginationPrimitive.Next
-                size="sm"
-                onClick={handleGoToPage(currentPage < totalPages ? currentPage + 1 : undefined)}
+                onClick={goToPage ? handleGoToPage(currentPage < totalPages ? currentPage + 1 : undefined) : undefined}
+                href={getPageLink?.(currentPage < totalPages ? currentPage + 1 : currentPage)}
                 disabled={currentPage === totalPages}
                 t={t}
               />
             </PaginationPrimitive.Item>
           </PaginationPrimitive.Content>
         ) : (
-          <PaginationPrimitive.Content>
+          <PaginationPrimitive.Content className="cn-pagination-hide-pages">
             {/* Previous Button */}
-            <PaginationPrimitive.Item>
+            <PaginationPrimitive.Item className="cn-pagination-item-previous">
               <PaginationPrimitive.Previous
-                size="sm"
-                onClick={handleGoToPage(previousPage ?? undefined)}
-                disabled={!previousPage}
+                href={getPrevPageLink?.()}
+                onClick={onPrevious}
+                disabled={!hasPrevious}
                 t={t}
               />
             </PaginationPrimitive.Item>
             {/* Next Button */}
-            <PaginationPrimitive.Item>
-              <PaginationPrimitive.Next
-                size="sm"
-                onClick={handleGoToPage(nextPage ?? undefined)}
-                disabled={!nextPage}
-                t={t}
-              />
+            <PaginationPrimitive.Item className="cn-pagination-item-next">
+              <PaginationPrimitive.Next href={getNextPageLink?.()} onClick={onNext} disabled={!hasNext} t={t} />
```

Comment:
On a indeterminate variant, I think we can make `getNextPageLink` and `getPrevPageLink` required props

Reply:
We saw a scenario (in pipeline studio steps palette) where we used `onNext` and `onPrevious` ignoring `getPrevPageLink ` and `getNextPageLink`. So either we can pass callbacks or lint getter functions.